### PR TITLE
Add upi/gcp/rhel8 and microshift e2e workflow 

### PR DESCRIPTION
**Pull request by sallyom**
This PR adds a workflow that will provision and deprovision a gcp centos8 machine. 
TODO: replace centos8 with rhel8 - need to get the subscription-manager worked out. 
Also, a workflow for installing and starting microshift in the gcp instance.

### CODE REVIEW

File: ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml [hunk 1]
```diff
@@ -49,16 +56,35 @@ images:
...
+        source_path: /etc/yum.repos.d/google-cloud-sdk.repo
+  to: microshift-setup
 resources:
   '*':
     limits:
       memory: 4Gi
     requests:
       cpu: 100m
       memory: 200Mi
+tag_specification:
```
Comment (copejon) [inline ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml:85]: This I thought declared the microshift images as a component of OCP, where as promotion would isolate it from OCP releases.  Does it matter here which we use?

File: ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml [hunk 2]
```diff
@@ -1,26 +1,33 @@
 base_images:
   base:
-    name: "4.7"
+    name: "4.8"
     namespace: ocp
     tag: base
+  libvirt-installer:
```
Comment (copejon) [inline ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml:6]: Is this effectively a dupe of `base`?

File: ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml [hunk 3]
```diff
@@ -49,16 +56,35 @@ images:
...
 resources:
   '*':
     limits:
       memory: 4Gi
     requests:
       cpu: 100m
       memory: 200Mi
+tag_specification:
+  name: "4.8"
+  namespace: ocp
```
Comment (copejon) [inline ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml:87]: This I thought declared the microshift images as a component of OCP, where as promotion would isolate it from OCP releases. Does it matter here which we use?

Comment (sallyom) [reply ci-operator/config/redhat-et/microshift/redhat-et-microshift-main.yaml:91]: you are right, yes, this afaict isn't necessary, and you'll want to keep the promotion you have in your PR, I was trying to isolate a 'missing dependency in step' message I kept getting.

File: ci-operator/step-registry/redhat-et/microshift/e2e/test/redhat-et-microshift-e2e-test-commands.sh [hunk 4]
```diff
@@ -170,21 +163,12 @@ get_kubectl() {
...
-    if [ $ARCH = "x86_64" ]; then
-        curl -L https://github.com/redhat-et/microshift/releases/download/$VERSION/microshift-linux-amd64 -o microshift
-        curl -L https://github.com/redhat-et/microshift/releases/download/$VERSION/release.sha256 -o release.sha256
-    fi
-
-    SHA=$(sha256sum microshift | awk '{print $1}')
-    if [[ $SHA != $(cat release.sha256 | awk '{print $1}') ]]; then echo "SHA256 checksum failed" && exit 1; fi
-
     sudo chmod +x microshift
     sudo mv microshift /usr/local/bin/
```
Comment (sallyom) [inline ci-operator/step-registry/redhat-et/microshift/e2e/test/redhat-et-microshift-e2e-test-commands.sh:167]: I think I saw an error here on the last run, if so, could try
`sudo mv $HOME/microshift /usr/local/bin/`

**Comment by stevekuznetsov** [comment]
:(

/approve

**Pull request merged**